ait/core/server/handler.py

```diff
@@ -0,0 +1,106 @@
+import __builtin__
+
+from ait.core import log, tlm
+
+
+class Handler(object):
+    """
+    This is the base Handler class that all custom handlers must inherit
+    from. All custom handlers must implement the handle method, which will
+    called by the stream the handler is attached to when the stream receives
+    data.
+    """
+
+    def __init__(self, input_type=None, output_type=None, **kwargs):
+        self.input_type = input_type
+        self.output_type = output_type
+
+        for key, value in kwargs.items():
+            setattr(self, key, value)
+
+    def __repr__(self):
+        return '<handler.%s>' % (self.__class__.__name__)
+
+    def validate_input(self, input_data):
+        """
+        Attempts to convert input data to expected input type, if specified.
+        If no input type specified, returns input data.
+        If conversion fails, raises Exception.
+        If conversion successful, returns converted data.
+        """
+        if self.input_type:
+            try:
+                converted = getattr(__builtin__,
+                                    self.input_type.decode())(input_data)
+            except Exception as e:
+                raise(e)
+
+            return converted
+
+        return input_data
+
+    def execute_handler(self, input_data):
+        """
+        Checks that the input data to the handler matches the expected
+        input type. If input data is of the correct type, calls the
+        handle method. If input data is not the correct type, raises a
+        ValueError.
+        """
+        try:
+            checked_input = self.validate_input(input_data)
+        except Exception as e:
+            raise(ValueError('Input is not of valid type: ', e))
```

Comment:
Are we catching this `ValueError` when we pass data through handlers in our streams? We should make sure that's resilient to issues (regardless of the `validate_input`)

Reply:
+1 need to catch and log this error at the stream level

Reply:
(and this should not be a generic exception catch that then raises a ValueError)

Reply:
Removed `vaildate_input` functionality